### ISSUE TITLE
Monkey Patching is slow

### ISSUE THREAD

**Issue by DakuTree** [DakuTree/manga-tracker#76]
At the moment I'm using Monkey Patching to properly test when sites fail, the problem is it more than doubles the amount of time required to test.

Last build before enabling (1min): https://travis-ci.org/DakuTree/manga-tracker/builds/179898586
First build after enabling (2min 24s): https://travis-ci.org/DakuTree/manga-tracker/builds/181030365
Looking at some of the later builds, the time ranges between 2min20s & 4m which is a bit ugh. If I was using it more, then the time increase would be fair, but I'm barely using it.

Basically, I'd like to turn off Monkey Patching. The problem is its very difficult to replicate the functionality without a bunch of rewriting.
The only thing I'm rewriting is log_message, which I then check after calling getTitleData and verifying it ran & making sure it returned the correct error message.
I was thinking about poking about with Exceptions which _I'm assuming_ can be tweaked to also send a log_message as well. We can then just verify failures via checking the exception message.
